drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/analysis/controller/Events.java

```diff
@@ -43,21 +43,23 @@ public Events( final EventBus eventBus,
     }
 
     public void setup() {
-
-        eventBusHandlerRegistrations.add( eventBus.addHandler( ValidateEvent.TYPE,
-                                                               analyzerController ) );
-        eventBusHandlerRegistrations.add( eventBus.addHandler( DeleteRowEvent.TYPE,
-                                                               analyzerController ) );
-        eventBusHandlerRegistrations.add( eventBus.addHandler( AfterColumnDeleted.TYPE,
-                                                               analyzerController ) );
-        eventBusHandlerRegistrations.add( eventBus.addHandler( UpdateColumnDataEvent.TYPE,
-                                                               analyzerController ) );
-        eventBusHandlerRegistrations.add( eventBus.addHandler( AppendRowEvent.TYPE,
-                                                               analyzerController ) );
-        eventBusHandlerRegistrations.add( eventBus.addHandler( InsertRowEvent.TYPE,
-                                                               analyzerController ) );
-        eventBusHandlerRegistrations.add( eventBus.addHandler( AfterColumnInserted.TYPE,
-                                                               analyzerController ) );
+        if ( canSetUp ) {
```

Comment:
Yeah this works doesn't it... it may have been nicer to record the ```HandlerRegistration``` from the ```add(...)``` method and only reattach the handler if the registration is ```null```.. still your change works..